### PR TITLE
[tr064] Improvements of Phonebook Profile

### PR DESCRIPTION
**Pull request by cweitkamp**
- Improvements of Phonebook Profile

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfileFactory.java [hunk 1]
```diff
@@ -58,12 +81,37 @@
...
+        final ProfileType localizedProfileType = localize(profileType, locale);
+        if (localizedProfileType != null) {
+            localizedProfileTypeCache.put(localizedKey, localizedProfileType);
+            return localizedProfileType;
+        } else {
+            return profileType;
+        }
+    }
+
+    private @Nullable ProfileType localize(ProfileType profileType, @Nullable Locale locale) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfileFactory.java:109]: Why do you need these two methods? If I didn't overlook something, `localize` and `getLocalizedProfileTypeKey` are only called once.

**Review by J-N-K — COMMENTED** [review]
Othewrwise LGTM

File: bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfileFactory.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfileFactory.java]: Absolutely right. It is a developer's sin. Copy & paste from OHC `SystemProfileFactory`.

**Review by cpmeister — APPROVED** [review]
LGTM

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
Unfortunately it fails with the tado lib, just before we get to the interesting part of the build.

**Comment by cweitkamp** [comment]
I pushed a new version including support for resolving CallItem states (StringListTypes). Maybe Jenkins will now be happy.

**Comment by cweitkamp** [comment]
I did some more changes and finally got it working perfectly. Sry for the mass of commits in between.

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
Please run spotless. Otherwise LGTM.

**Comment by cweitkamp** [comment]
@J-N-K Done.

**Pull request merged**